src/core/datatypes/phylogenetics/TreeDiscreteCharacterData.cpp

```diff
@@ -112,9 +112,12 @@ void TreeDiscreteCharacterData::initFromString(const std::string &s)
     throw RbException("Cannot initialize a tree with a discrete character data matrix from a string.");
 }
 
-
 void TreeDiscreteCharacterData::setCharacterData( AbstractHomologousDiscreteCharacterData *d )
 {
+	if ( character_data != NULL )
```

Comment:
You don't need the if-statement here, I believe -- `delete` can handle `NULL`.
Also, you can use `nullptr` now, instead of `NULL`, which is a macro.

Reply:
Good to know! Fixed.